<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>521</width>
    <height>606</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Knight's Tour Solver</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QFormLayout" name="formLayout">
      <property name="labelAlignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="formAlignment">
       <set>Qt::AlignHCenter|Qt::AlignTop</set>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="chessboardSizeLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Chessboard size (5 - 20)</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="chessboardSizeSpinBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="wrapping">
         <bool>false</bool>
        </property>
        <property name="frame">
         <bool>true</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::UpDownArrows</enum>
        </property>
        <property name="minimum">
         <number>5</number>
        </property>
        <property name="maximum">
         <number>20</number>
        </property>
        <property name="value">
         <number>8</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QGraphicsView" name="chessboardView">
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="dragMode">
       <enum>QGraphicsView::NoDrag</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="searchForSolutionCancelOperationButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Search for solution</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QStackedWidget" name="solutionStackedWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="page">
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QProgressBar" name="solutionProgressBar">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximum">
               <number>0</number>
              </property>
              <property name="value">
               <number>0</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2">
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="margin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="solutionLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QRadioButton" name="showSolutionNumbersRadioButton">
          <property name="text">
           <string>Show solution numbers</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="showSolutionLinesRadioButton">
          <property name="text">
           <string>Show solution lines</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="backButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Back</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>chessboardSizeSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>resetChessboard(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>23</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchForSolutionCancelOperationButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSearchForSolutionCancelOperationButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>627</x>
     <y>543</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showSolutionNumbersRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onShowSolutionNumbersRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>614</x>
     <y>523</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showSolutionLinesRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onShowSolutionLinesRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>551</y>
    </hint>
    <hint type="destinationlabel">
     <x>557</x>
     <y>553</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onBackButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>404</x>
     <y>582</y>
    </hint>
    <hint type="destinationlabel">
     <x>606</x>
     <y>584</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>resetChessboard(int)</slot>
  <slot>onSearchForSolutionCancelOperationButtonClicked()</slot>
  <slot>onShowSolutionNumbersRadioButtonToggled(bool)</slot>
  <slot>onShowSolutionLinesRadioButtonToggled(bool)</slot>
  <slot>onBackButtonClicked()</slot>
 </slots>
</ui>
